src/linker/Linker.Steps/SubStepsDispatcher.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+using Mono.Cecil;
+using Mono.Collections.Generic;
+
+namespace Mono.Linker.Steps
+{
+	//
+	// Generic steps dispatcher is intended to by used by custom linker step which
+	// consist of multiple steps. It simplifies their implementation as well as the
+	// way how to hook them into the pipeline of existing steps.
+	//
+	public abstract class SubStepsDispatcher : IStep, IEnumerable<ISubStep>
```

Comment:
Also curious: What is the use case for having the `IEnumerable<ISubStep>` implemented by the class?

Reply:
Especially since we don't expose that through the ref assemblies.

Reply:
Removed